server/job_stream.go

```diff
@@ -29,6 +29,10 @@ import (
 
 var getNewTokenTimeout = 3 * time.Second
 
+// startStreamProcessingFunc is an alias for startStreamProcessing that can be overridden in tests
+// to avoid starting up actual stream processing
+var startStreamProcessingFunc = startStreamProcessing
```

Comment:
FWIW, we have the same [pattern in discovery](https://github.com/livepeer/go-livepeer/blob/1b0e740345a1b44508749225e123d63c735c8948/discovery/discovery.go#L30-L31) and it's been horrible for test correctness, I've been trying to [unwind](https://github.com/livepeer/go-livepeer/pull/3798) our use of that so we can have cleaner tests that pass more reliably.

If it's not easy to override whatever the blocker is here then I'd suggest putting `startStreamProcessingFunc` in some struct that can be set on a per-test basis with `startStreamProcessing` as the default for normal use.